lib/spack/spack/build_systems/autotools.py

```diff
@@ -291,49 +291,112 @@ def check(self):
             self._if_make_target_execute('test')
             self._if_make_target_execute('check')
 
-    def _activate_or_not(self, active, inactive, name, active_parameters=None):
+    def _activate_or_not(self, active, inactive, name, activation=None):
+        """This function contains the current implementation details of
+        :py:meth:`~.AutotoolsPackage.with_or_without` and
+        :py:meth:`~.AutotoolsPackage.enable_or_disable`.
+
+        Args:
+            active (str): the default activation word ('with' in the case
+                of ``with_or_without``)
+            inactive (str): the default de-activation word ('without' in
+                the case of ``with_or_without``)
+            name (str): name of the variant that is being processed
+            activation (callable): callable that accepts a single
+                value and returns the parameter to be used leading to an entry
+                of the type '--{active}-{name}={parameter}
+
+        Returns:
+            list of strings that corresponds to the activation/deactivation
+            of the variant that has been processed
+        """
         spec = self.spec
         args = []
+
+        if activation == 'prefix':
+            activation = lambda x: spec[x].prefix
```

Comment:
This is just syntactic sugar so the user doesn't have to input a function, right?